sampledb/logic/schemas/validation_preprocessor.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov 26 15:32:24 2020
+
+@author: b.mayer
+
+This validation preprocessor extends submitted update data silently if possible.
+So that partial updates or redundant field  definitions like pint's dimensionality do not have to be submitted everytime. 
+"""
+
+import flask
+
+from ...logic import objects, datatypes, errors, units
+from .utils import units_are_valid, get_dimensionality_for_units
+
+def _validation_preprocessor_object(instance: dict, schema: dict, object_id: int) -> None:
+    """
+    If the given object is the root object to be updated 
+    (not a data type "object" reference to another object)
+    then this function extends the data in the instance dictionary 
+    in case there are any schema properties missing with the data 
+    from the objects current version. This allows for partial updates 
+    via the API while keeping the rest of the data untouched.
+    
+    This function can be toggled on via the configuration parameter SAMPLEDB_ALLOW_PARTIAL_OBJECT_UPDATES
+
+    :param instance: the sample object
+    :param schema: the valid sampledb object schema
+    """
+    if flask.current_app.config['ALLOW_PARTIAL_OBJECT_UPDATES'] is False:
```

Comment:
Why not test for truthiness, e.g. `if not flask.current_app.config['ALLOW_PARTIAL_OBJECT_UPDATES']:`?